### DISCUSSION TITLE
dom.log

### DISCUSSION THREAD

**Discussion by chdelfosse** [epeios-q37/atlas-python#14] [Ideas]
dom.log(x) takes a single string argument.
if JSON were applied to that single argument, dom.log could be used in all cases where
the argument does not provide a `__str__` method (when using `"%s"%argument` ).

I think that this is more useful than having dom.log() taking any number  of arguments.

**Comment by epeios-q37** [comment]
I'm not very familiar with JSON, and I've never used it with Python, so can you show me an example of how `dom.log()` should work with JSON?

**Comment by chdelfosse** [comment]
If the argument `x` received by `dom.log` is passed to `JavaScript` JSON.stringify(x), it becomes a string however complex it might be, or remains a string if it was a string in the first place.

If this were to become the implementation, it would have to be [easily] documented,
and since a string remains a string it would be backwards compatible.

On the other hand the documentation could just (should?) point out that the argument to the `dom.log` function MUST be a string,
and that python's `"%s"%<object>"` fails if the object does not have a `__str__`  method,
(unless it is obvious),
but that `dom.log("%s"%json.dumps(<object>))` always yields a usable result
(ditto).

I do not know the implications for the other languages that `atlastk` supports, and therefore this discussion may be closed from my point of view.
This is about python, I will always use `json.dumps`  with dom.log() .
